client/core/core.go

```diff
@@ -854,18 +862,21 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 	log.Tracef("processing tip change for %s", unbip(assetID))
 	c.waiterMtx.Lock()
 	defer c.waiterMtx.Unlock()
-	for _, w := range c.waiters {
+	for d, w := range c.waiters {
 		waiter := w
 		if waiter.asset.ID != assetID {
 			continue
 		}
+		dex := d
 		go func() {
 			confs, err := waiter.coin.Confirmations()
```

Comment:
Seems like you could put `defer c.removeWaiter(dex)` as the first line in this goroutine func.

Reply:
In the case of no error, but too few confirmations, the waiter should remain in the map.

Reply:
Ah, gotcha.